### PR TITLE
Bump bdk to 0.20

### PR DESCRIPTION
**Pull request by bonomat**

### CODE REVIEW

**Comment by klochowicz** [comment]
I'm afraid that in order to preserve backwards compatibility, this should be against `maia-0.1` branch, along with release 0.1.2 of maia.
The next step would be merging that back into `main`, with `0.2.2` maia release.

Only in this way we can keep the compatibility of both versions having same `maia-core` version. 

Are you OK for me to rework this PR to do it?

File: maia-core/src/interval.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ const MAX_DIGITS: usize = 20;
 const BASE: usize = 2;
 
 /// Binary representation of a price interval.
-#[derive(Clone, Debug, PartialEq)]
```
Comment (klochowicz) [inline maia-core/src/interval.rs:17]: 🤔a new clippy lint from new rustc version?

**Comment by bonomat** [comment]
Before we do this let's rethink if we need to keep backwards compatibility or if we can avoid this.

File: maia-core/src/interval.rs (hunk 1 above)
Comment (bonomat) [reply maia-core/src/interval.rs:17]: This is only for `beta` target. I'm not sure why we even build it for `beta` 😅

**Comment by klochowicz** [comment]
> Before we do this let's rethink if we need to keep backwards compatibility or if we can avoid this.

it's pretty trivial to do, so I'd roll it out, and only stop supporting maia 0.1.x when we explicitly say so in itchysats - I expect this to take a bit of time, but I'd vouch for this being the last attempt of doing it.

I would like to decouple bdk update from that decision, so it doesn't put more pressure on bringing mandatory update forward.

a bit of context: the time required to preserve backwards-compatibility is around 10 minutes.

**Comment by bonomat** [comment]
I disagree here as we jump too fast on new versions without knowing what we get from them and we can think twice before we do that. 

Also, yesterday in the call we shortly addressed that we want to drop support to 0.4.16 aka _the old connection_ which would remove the need of keeping 2 maia versions in itchysats and also remove the need for backwards compatibility.

File: maia-core/Cargo.toml [hunk 2]
```diff
@@ -1,13 +1,13 @@
 [package]
 name = "maia-core"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (luckysori) [inline maia-core/Cargo.toml:3]: ❓ Is this not a breaking change? I'm assuming that anyone using `maia-core` will need to upgrade to `bdk:0.21`.

**Comment by klochowicz** [comment]
Superceded by #59

**Pull request closed without merging**